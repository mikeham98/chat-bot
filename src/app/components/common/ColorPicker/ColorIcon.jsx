import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import styles from '../../../../themes/components/colorPicker.scss';

export default class ColourIcon extends React.PureComponent {
    returnClassName() {
        let className = [styles.colourIconWrapper];
        if(this.props.selected) {
            className.push(styles.colourIconSelected)
        }
        return classnames(className);
    }

    render() {
        return (
            <div
                style={{backgroundColor: this.props.color}}
                onClick={this.props.onClick}
                className={this.returnClassName()}
            />
        )
    }
}

ColourIcon.propTypes = {
    color: PropTypes.string,
    selected: PropTypes.bool,
    onClick: PropTypes.func.isRequired,
};