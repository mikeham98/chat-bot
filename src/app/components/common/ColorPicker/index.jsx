import React from 'react';
import PropTypes from "prop-types";
import styles from '../../../../themes/components/colorPicker.scss';
import {flatColours} from '../../../config/colours.config';
import ColourIcon from "./ColorIcon";

export default class ColorPicker extends React.PureComponent {
    returnColorIcons() {
        return flatColours.map((color) => {
            return (
                <ColourIcon
                    key={color}
                    selected={color === this.props.color}
                    color={color}
                    onClick={() => this.props.onChange(color)}
                />
            )
        });
    }

    render() {
        return (
            <div className={styles.colorPickerWrapper}>
                {this.returnColorIcons()}
            </div>
        )
    }
}

ColorPicker.propTypes = {
    flatColours: PropTypes.arrayOf(PropTypes.string),
    onChange: PropTypes.func.isRequired,
};