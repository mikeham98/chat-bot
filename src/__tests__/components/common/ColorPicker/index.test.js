import React from 'react';
import {shallow} from 'enzyme';
import ColorPicker from '../../../../app/components/common/ColorPicker/index';
import ColorIcon from '../../../../app/components/common/ColorPicker/ColorIcon';

jest.mock('../../../../app/config/colours.config', () => ({
    flatColours: ['red', 'orange', 'yellow','green']
}));

describe('ColorPicker', () => {
    let shallowedWrapper;
    let props = {};
    const wrapper = () => {
        if (!shallowedWrapper) {
            shallowedWrapper = shallow(<ColorPicker {...props}/>);
        }
        return shallowedWrapper;
    };

    beforeEach(() => {
        props = {
            color: 'orange',
            onChange: jest.fn(),
        };
        shallowedWrapper = undefined;
    });

    describe('render', () => {
        it('should return four ColorIcon components with the orange ColorIcon selected', () => {
            expect(wrapper()).toMatchSnapshot();
        });
    });

    describe('functionality', () => {
        describe('prop', () => {
            describe('onChange', () => {
                it('should be called on with yellow onClick of the orange ColorIcon', () => {
                    expect(props.onChange).not.toHaveBeenCalled();
                    wrapper().find(ColorIcon).at(2).props().onClick();
                    expect(props.onChange).toHaveBeenCalledTimes(1);
                });
            });
        });
    });
});