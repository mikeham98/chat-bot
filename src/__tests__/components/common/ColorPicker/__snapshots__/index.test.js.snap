// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorPicker render should return four ColorIcon components with the orange ColorIcon selected 1`] = `
<div
  className="colorPickerWrapper"
>
  <ColourIcon
    color="red"
    key="red"
    onClick={[Function]}
    selected={false}
  />
  <ColourIcon
    color="orange"
    key="orange"
    onClick={[Function]}
    selected={true}
  />
  <ColourIcon
    color="yellow"
    key="yellow"
    onClick={[Function]}
    selected={false}
  />
  <ColourIcon
    color="green"
    key="green"
    onClick={[Function]}
    selected={false}
  />
</div>
`;
